src/main/java/net/nextencia/rrdiagram/BNFProcessor.java

```diff
@@ -0,0 +1,259 @@
+package net.nextencia.rrdiagram;
+
+import net.nextencia.rrdiagram.common.Utils;
+import net.nextencia.rrdiagram.grammar.model.*;
+import net.nextencia.rrdiagram.grammar.rrdiagram.RRDiagram;
+import net.nextencia.rrdiagram.grammar.rrdiagram.RRDiagramToSVG;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.Executors;
+import java.util.logging.Logger;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.HashMap;
+
+class BNFProcessor {
+    static Logger logger = Logger.getLogger(BNFProcessor.class.getName());
```

Comment:
Is the indentation really supposed to be 4-space?

Reply:
that was my default . switched to 2